noria/src/data.rs

```diff
@@ -561,6 +561,46 @@ impl<'a> From<&'a str> for DataType {
     }
 }
 
+impl TryFrom<mysql::Value> for DataType {
+    type Error = &'static str;
+
+    fn try_from(v: mysql::Value) -> Result<Self, Self::Error> {
+        match v {
+            mysql::Value::NULL => Ok(DataType::None),
+            mysql::Value::Bytes(v) => {
+                let len = v.len();
```

Comment:
Is it possible to avoid duplicating this code (which also exists elsewhere for `DataType`) here, and instead just calling the other impl that already has it?

Reply:
I changed it so as to not duplicate, but the `Value::Bytes` branch is not perfect. I now reuse the `From<String>` impl, but that converter is in fact not safe (with strings containing null-characters), so I added a check in my converter.